modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -3434,8 +3434,11 @@ def _compute_new_widths():
         if new_lengths is None:
             new_lengths = new_lengths2
         new_dtypes = None
+        new_index = None
+        new_columns = None
         if axis == Axis.ROW_WISE:
-            new_index = self.index.append([other.index for other in others])
+            if all((obj.has_materialized_index for obj in (self, *others))):
+                new_index = self.index.append([other.index for other in others])
```

Comment:
```suggestion
            if all(obj.has_materialized_index for obj in (self, *others)):
                new_index = self.index.append(other.index for other in others)
```
let's use generator form where possible, it's usually taking less memory and is slightly faster as we don't have to build an extra intermediate Python object

Reply:
`append` doesn't accept generators:
https://github.com/pandas-dev/pandas/blob/v2.0.3/pandas/core/indexes/base.py#L5230-L5258